tests/runProviderTest.ts

```diff
@@ -1,9 +1,13 @@
 import { JoshProvider, JoshProviderError, MathOperator, Method, Payload } from '@joshdb/core';
 
-export default (Provider: typeof JoshProvider, ProviderError: typeof JoshProviderError, providerData: unknown) => {
-	describe('Provider', () => {
+export function runProviderTest<ProviderOptions extends JoshProvider.Options>(
+	Provider: typeof JoshProvider,
+	ProviderError: typeof JoshProviderError,
+	providerOptions: ProviderOptions
+) {
```

Comment:
This should hopefully fix all of those `@ts-expect-error`s.

```suggestion
export function runProviderTest<Provider extends JoshProvider, ProviderOptions extends JoshProvider.Options>(
	Provider: typeof Provider,
	ProviderError: typeof JoshProviderError,
	providerOptions: ProviderOptions
) {
```